### ISSUE TITLE
SvelteKit ran with Netlify CLI fails to start

### ISSUE THREAD

**Issue by brittneypostma** [netlify/cli#5131]
### Describe the bug

When running `ntl dev` in a SvelteKit project, the server fails to start because it is looking for port `3000`. SvelteKit updated to run `vite dev` and is now running on port `5173` by default. This can be fixed by adding a `targetPort` to the netlify.toml file, but should run port 5173 by default.

### Steps to reproduce

1. Run git clone git@github.com:brittneypostma/sveltekit-cli-bug.git
2. Navigate to repo
3. Run ntl dev
4. Enter on either SvelteKit or Vite command to run `npm run dev`
5. See error `vite: not found` `Waiting for framework port 3000. This can be configured using the 'targetPort' property in the netlify.toml◈ Command failed with exit code 127: npm run dev. Shutting down Netlify Dev server`

## Fix
Port is now 5173, not 3000

## Temporary fix
1. Add `targetPort` to netlify.toml file
```
[dev]
  targetPort = 5173
```
2. Rerun `ntl dev`
3. Server starts

### Configuration

n/a

### Environment

```
  System:
    OS: Linux 5.10 Ubuntu 20.04 LTS (Focal Fossa)
    CPU: (16) x64 Intel(R) Core(TM) i7-10700KF CPU @ 3.80GHz
    Memory: 13.97 GB / 15.57 GB
    Container: Yes
    Shell: 5.0.16 - /bin/bash
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
  npmGlobalPackages:
    netlify-cli: 11.8.2
```

![image](https://user-images.githubusercontent.com/45889730/193837307-d6fbd184-99f8-43f2-a01f-bf3e01778531.png)

**Comment by danez** [comment]
This should work as of version 12.0.7

**Comment by SeytkhanB** [comment]
Oh my god! You saved me. thanks a lot!

**Comment by kavfam** [comment]
Nov 24 2022. Using Node 16.14.0,  vite 3.3.2, Svelte 3.52.0. netlify.toml file has [dev] targetPort = 5173
1. netlify dev with Svelte 'npm run dev' gives indefinite "Waiting for framework port 5173".
2. netlify dev --targetPort 5173 with Svelte 'npm run dev' gives indefinite "Waiting for framework port 5173".
3. Netlify dev with Vite 'npm run dev'  or "npm run build" works ie gives message "Server now ready on http://localhost:8088"

**Comment by kavfam** [comment]
ok I have this working now.  I decided to upgrade to Node v18.12.1 as I was getting messages about "module" which was updated since v17 and I was running v16.14.0. 
However to get netlify to not hang waiting for port 5173, I had to have my netlify.toml file like this:
[build]
  publish="dist"
[dev]
 command = "npm run dev"
  targetPort=5173
  port=8888
[functions]
node_bundler = "esbuild"

With those changes (possibly just the toml file) netlify dev runs and opens my svelte/vite app no problem.
